File: backend/app/schemas/activity.py
"""活动相关模式"""
from pydantic import BaseModel

ACTION_LABELS = {
    "create_kb": "创建了知识库",
    "upload_doc": "上传了文档",
    "add_member": "添加了成员",
    "create_note": "新建了笔记",
}


class ActivityResponse(BaseModel):
    id: int
    user_id: int
    username: str
    action: str
    action_label: str
    knowledge_base_id: int | None
    knowledge_base_name: str | None
    knowledge_base_owner: str | None = None  # 知识库所有者，用于显示 owner/kb 格式
    extra: dict | None
    created_at: str

File: frontend/src/api/activity.ts
import client from "./client";

export interface Activity {
  id: number;
  user_id: number;
  username: string;
  action: string;
  action_label: string;
  knowledge_base_id: number | null;
  knowledge_base_name: string | null;
  knowledge_base_owner?: string | null;
  extra: Record<string, unknown> | null;
  created_at: string;
}

export const activityApi = {
  list: (scope: "all" | "mine" = "all", limit = 50) =>
    client.get<Activity[]>("/activities", { params: { scope, limit } }),
};
